neo/Ledger/MemoryPool.cs

```diff
@@ -314,14 +314,15 @@ internal bool TryAdd(UInt256 hash, Transaction tx)
 
         private void RemoveOverCapacity(ref List<Transaction> removedTransactions)
         {
+            if (removedTransactions == null)
+                removedTransactions = new List<Transaction>();
```

Comment:
The whole point of delaying the instantiation of this was to do it delayed only if it went in the while, since the majority of times this is called it will never enter the while at all. Anyway, it is changed now to return the removed transactions in the latest iteration of the PR.